cmd/kubeadm/app/cmd/util/cmdutil.go

```diff
@@ -68,8 +68,9 @@ func FindExistingKubeConfig(file string) string {
 		return file
 	}
 	// The user did not provide a --kubeconfig flag. Find a config in the standard
-	// locations using DefaultClientConfigLoadingRules, but also consider the default config path.
+	// locations using DefaultClientConfigLoadingRules, but also consider the default config path
+	// as the primary location to check.
 	rules := clientcmd.NewDefaultClientConfigLoadingRules()
-	rules.Precedence = append(rules.Precedence, kubeadmconstants.GetAdminKubeConfigPath())
+	rules.Precedence = append([]string{kubeadmconstants.GetAdminKubeConfigPath()}, rules.Precedence...)
```

Comment:
IMO the `KUBECONFIG` env var should take precedence over the default path, so we need to modify `NewDefaultClientConfigLoadingRules`

Reply:
that was my original thinking when i added this code as well.
but i think @fabriziopandini disagreed with that the last time we spoke about it.

we cannot easily modify `NewDefaultClientConfigLoadingRules` as it might break client-go for multiple consumers. we can hack around it though, by finding where the KUBECONFIG env var is in the slice and prepending. not a great hack.

Reply:
we can check if the first element of the slice is `RecommendedHomeFile` and if yes prepending the default path, if not append to the end of slice.

Reply:
ah, i guess this will work yes.
will still wait on @fabriziopandini to see what he thinks about "kubeadm default path vs KUBECONFIG env" as the first search location.

also:
/assign @timothysc